lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```
@@ -296,6 +298,55 @@ def self.settings_for_dependent_targets(target, dependent_targets)
           build_settings
         end
 
+        # Add custom build settings and required build settings to link to
```

copied pasted from `aggregate_xcconfig.rb`, almost zero changes except passing parameters.